src/detection/PsfFormatter.cc

```diff
@@ -22,14 +22,13 @@
 #include "lsst/daf/persistence/BoostStorage.h"
 #include "lsst/daf/persistence/XmlStorage.h"
 #include "lsst/pex/exceptions.h"
-#include "lsst/pex/logging/Trace.h"
+#include "lsst/log/Log.h"
 #include "lsst/pex/policy/Policy.h"
 
 BOOST_CLASS_EXPORT(lsst::afw::detection::Psf)
 
-#define EXEC_TRACE  20
-static void execTrace(std::string s, int level = EXEC_TRACE) {
-    lsst::pex::logging::Trace("afw.detection.PsfFormatter", level, s);
+namespace {
+LOG_LOGGER _log = LOG_GET("afw.detection.PsfFormatter");
```

Comment:
At first glance, it looks like this isn't safe: default-constructing a `Log` object invokes `lsst::log::log4cxxInit`, which checks the status of another global variable, `lsst::log::doBasicConfig`, which isn't guaranteed to have been initialized yet at that point.  I'm not *completely* positive that's a problem, because `doBasicConfig` is a bool, and `log4cxxInit` seems to do something sensible whether it's true or false (but its behavior does seem to depend on the undefined status of `doBasicConfig`, which is worrying).

You can fix this using the construct-on-first-use approach recommended in the link above, but I wonder if it would be possible to make it safe to construct `Log` instances at global initialization time, since a lot of people are probably going to try to do that.  Perhaps @andy-slac has an opinion on this?  It looks like there's already an attempt to make it safe to initialize loggers at global initialization time, and maybe I've just misread the code in thinking that it isn't.